packages/vm/src/context.rs

```diff
@@ -10,27 +10,29 @@ use wasmer_runtime_core::vm::Ctx;
 use cosmwasm_std::{Api, Binary, CanonicalAddr, HumanAddr, Storage};
 
 use crate::errors::Error;
-use crate::memory::{read_region, write_region};
+use crate::memory::{read_region, to_region, write_region};
 
 /// An unknown error occurred when writing to region
 static ERROR_WRITE_TO_REGION_UNKNONW: i32 = -1000001;
 /// Could not write to region because it is too small
 static ERROR_WRITE_TO_REGION_TOO_SMALL: i32 = -1000002;
 
+/// Reads a storage entry from the VM's storage into Wasm memory
 pub fn do_read<T: Storage>(ctx: &Ctx, key_ptr: u32, value_ptr: u32) -> i32 {
     let key = read_region(ctx, key_ptr);
     let mut value: Option<Vec<u8>> = None;
     with_storage_from_context(ctx, |store: &mut T| value = store.get(&key));
     match value {
         Some(buf) => match write_region(ctx, value_ptr, &buf) {
-            Ok(bytes_written) => bytes_written.try_into().unwrap(),
+            Ok(()) => to_region(ctx, value_ptr).length.try_into().unwrap(),
```

Comment:
why not just `buf.len().try_into().unwrap()`?

Reply:
That was what we were returning earlier: `Ok(data.len())` and this syntax is clearer

Reply:
Jap, good point, simpler.

But then: why should `env.read_db`, `env.canonicalize_address` and `env.humanize_address` return the value at all. It is written to the result region for read. The return value is alrady unused. We only check for errors (`< 0`). So we can return constant 0.

Reply:
Yes, we needed it before to truncate properly.

Now I agree, all these return values should only exist for -X error messages of 0 success (and leave +X undefined for a future idea)